auth-api/src/main/java/io/stargate/auth/api/resources/AuthResource.java

```diff
@@ -232,16 +250,50 @@ public Response createTokenFromUsername(
     } catch (UnauthorizedException e) {
       return Response.status(Response.Status.UNAUTHORIZED)
           .entity(new Error("Failed to create token: " + e.getMessage()))
+          .cacheControl(cacheControlNoCache())
           .build();
     } catch (Exception e) {
       logger.error("Failed to create token", e);
       return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
           .entity(new Error("Failed to create token: " + e.getMessage()))
+          .cacheControl(cacheControlNoCache())
           .build();
     }
 
     return Response.status(Response.Status.CREATED)
         .entity(new AuthTokenResponse().authToken(token))
+        .cacheControl(cacheControlWithAge())
         .build();
   }
+
+  /**
+   * Used to create a Cache-Control header containing the proper values for the token's max-age and
+   * s-maxage as defined by {@code stargate.auth_tokenttl}. If the property is not set then the
+   * default value of 1800 will be used.
+   *
+   * @return A {@link CacheControl} object which is converted into a Cache-Control header with
+   *     proper directives.
+   */
+  private CacheControl cacheControlWithAge() {
+    CacheControl cacheControl = new CacheControl();
+    cacheControl.setMaxAge(tokenMaxAge);
+    cacheControl.setSMaxAge(tokenMaxAge);
+
+    return cacheControl;
+  }
+
+  /**
+   * Used to create a Cache-Control header that will indicate the response should not be cached.
+   * This is intended to be used in the event of a failure to authenticate.
+   *
+   * @return A {@link CacheControl} object denoting the response should not be cached.
+   */
+  private CacheControl cacheControlNoCache() {
```

Comment:
Should probably be called cacheControlNoStore to avoid being confused that this is setting the `no-cache` header that has no guarantee to prevent caching. 🙂

Reply:
`clearCacheControl` or `emptyCacheControl` could be alternative names?